Attempt to resolve issue #527 where bitcoind returns html indicating …

Christewart opened the pull request:
…that the service is unavailable. We shouldn't throw an exception if that happens

This attempts to resolve #527 where bitcoind is not started yet, and returns a static html page saying status is unavailable. This should handle the html properly.

Christewart commented:
Looks like this is still happening as of 9b1e43b https://travis-ci.org/bitcoin-s/bitcoin-s/jobs/556239392#L651

Christewart commented:
So do you think there is a chance we are missing a `.map()` some where on `isStartedF`, where we are sending requests before `isStartedF` is completed?

torkelrogstad commented:
Since running Travis as sudo, I haven't seen the error (probably done 10 runs). However, this error has been popping up in about 50% of the runs: https://travis-ci.org/bitcoin-s/bitcoin-s/jobs/560497406#L1071

torkelrogstad committed:
Tweak error logging of bitcoind responses

In this commit we move the error handling part of bitcoind responses
closer to the source of where they will happen, hopefully
giving better error messages.
